pandas/core/strings.py

```diff
@@ -577,6 +578,12 @@ def str_replace(arr, pat, repl, n=-1, case=None, flags=0, regex=True):
         if callable(repl):
             raise ValueError("Cannot use a callable replacement when "
                              "regex=False")
+        # if regex is default None, and a single special character is given
+        # in pat, still take it as a literal, and raise the Future warning
+        if regex is None and len(pat) == 1 and re.findall(r"\W", pat):
```

Comment:
I don't understand the `'\W'` here. I think that's not quite right for non-word characters that are not regex symbols. Something like `@`

Does python have a list of regex symbols?

Is the heuristic `re.escape(pat) == pat` good enough? 

```python
In [46]: re.escape('@')
Out[46]: '@'

In [47]: re.escape('^')
Out[47]: '\\^'

```

Reply:
@TomAugspurger oh, indeed it's wrong! nice catch!! thanks for the correction!!

Reply:
sure! @TomAugspurger

Reply:
ehh, @TomAugspurger  escape somehow changed... wired
i did get same output the first time i tested on my jupyter notebook:
re.escape("@") -> "@"
but now, i cannot replicate this and get re.escape("@") -> "\\@" ...
so i will go for this option:
pat in list("[\^$.|?*+()]")